src/utils/types.ts

```diff
@@ -1,37 +1,32 @@
 import { createContext } from "react";
-import {
-  CONFIG_CATEGORY_TO_GROUP,
-  CONFIG_GROUP_STRUCTURE,
-  CONFIG_GROUP_TO_TYPE,
-} from "./configurations";
 import { types, Instance } from "mobx-state-tree";
+import { MONTHS, GROUP_STRUCTURE } from "./configurations";
 
-// MobX types
+// Model: Transactions
 export const Transaction = types
   .model("Transaction", {
     id: types.identifier,
     date: "", // DD.MM.YYYY
     description: "",
     category: "No category",
+    group: "No group",
+    type: types.optional(
+      types.union(types.literal("costs"), types.literal("revenues")),
+      "costs",
+    ),
     amount: 0,
   })
   .views((self) => ({
     get month() {
       return parseInt(self.date.split(".")[1]);
     },
-    get group() {
-      return CONFIG_CATEGORY_TO_GROUP[self.category];
-    },
-    get costOrRevenue() {
-      return CONFIG_GROUP_TO_TYPE[CONFIG_CATEGORY_TO_GROUP[self.category]]; // ??? Can we reuse the group view?
-    },
   }));
 
 export interface ITransaction extends Instance<typeof Transaction> {}
 
 export const TransactionStore = types
   .model("TransactionStore", {
-    transactions: types.optional(types.map(Transaction), {}),
+    transactions: types.map(Transaction),
```

Comment:
Je mettrais la configuration des groupes dans ce model, et définirait les groupes comme des références

Reply:
Understood, to be updated in future model refactoring.